includes/cache/wikia/CeleryPurge.class.php

```diff
@@ -71,6 +71,30 @@ static function onRestInPeace() {
 	}
 
 	/**
+	 * SUS-81: allow CDN purging by surrogate key
+	 *
+	 * Use OutputPage::tagWithSurrogateKeys() helper to emit proper headers
+	 *
+	 * @param string $key surrogate key to purge
+	 * @param string $service Fastly's service name (defaults to mediawiki)
+	 */
+	static function purgeBySurrogateKey( $key, $service = 'mediawiki' ) {
+		$caller = self::getPurgeCaller();
+		wfDebug( "Purging backtrace: " . wfGetAllCallers( false ) . "\n" );
+
+		WikiaLogger::instance()->info( 'varnish.purge', [
+			'key' => $key,
+			'service' => $service
+		] );
+
+		( new AsyncCeleryTask() )
+			->taskType( 'celery_workers.purger.purge' )
```

Comment:
This should probably be a task in `\Wikia\Tasks\Tasks`. Something like

```php
( new EdgeCachePurger( [ $key ], $service ) )->queue();
```
This would encapsulate the constants and the sequence of method calls and also make this operation more `grep` friendly.